packages/core/src/lib/lit-core.ts

```diff
@@ -751,15 +766,65 @@ export class LitCore {
         this.latestBlockhash = blockHashBody.blockhash;
         this.lastBlockHashRetrieved = Date.now();
         log('Done syncing state new blockhash: ', this.latestBlockhash);
+
+        if (!this.latestBlockhash) {
+          logError(
+            `Error getting latest blockhash from the nodes. Received: "${this.latestBlockhash}". Attempting to fetch blockhash manually using ethers with fallback RPC URLs...`
+          );
+          const provider = await this._getProviderWithFallback();
+
+          if (!provider) {
+            logError(
+              'All fallback RPC URLs failed. Unable to retrieve blockhash.'
+            );
+            return;
+          }
+
+          try {
+            const latestBlock = await provider.getBlock('latest');
+            this.latestBlockhash = latestBlock.hash;
+            this.lastBlockHashRetrieved = Date.now();
+            log(
+              'Successfully retrieved blockhash manually: ',
+              this.latestBlockhash
+            );
+          } catch (ethersError) {
+            logError('Failed to manually retrieve blockhash using ethers');
+          }
+        }
       })
-      .catch((err: BlockHashErrorResponse) => {
+      .catch(async (err: BlockHashErrorResponse) => {
         // Don't let error from this setInterval handler bubble up to runtime; it'd be an unhandledRejectionError
         logError(
-          'Error while attempting fetch new latestBlockhash:',
+          'Error while attempting to fetch new latestBlockhash:',
           err.messages,
-          'reason: ',
+          'Reason: ',
           err.reason
         );
+
+        log(
+          'Attempting to fetch blockhash manually using ethers with fallback RPC URLs...'
+        );
+        const provider = await this._getProviderWithFallback();
+
+        if (!provider) {
+          logError(
+            'All fallback RPC URLs failed. Unable to retrieve blockhash.'
+          );
+          return;
+        }
+
+        try {
+          const latestBlock = await provider.getBlock('latest');
+          this.latestBlockhash = latestBlock.hash;
+          this.lastBlockHashRetrieved = Date.now();
+          log(
+            'Successfully retrieved blockhash manually: ',
+            this.latestBlockhash
+          );
+        } catch (ethersError) {
+          logError('Failed to manually retrieve blockhash using ethers');
+        }
```

Comment:
Why is this duplicated here? If the provider fails it will likely fail again and originally it was fine not doing anything anyway. Maybe the `then` one is the extra one and it should throw when `!this.latestBlockhash`?


BTW, can't we remove the `_networkSyncInterval`? We are now always calling `getLatestBlockhash` instead of accessing `latestBlockhash` from outside. We could [make it `protected` or `private`](https://github.com/LIT-Protocol/js-sdk/pull/584#issuecomment-2278550582) in v7